backend/routes/api/resources.js

```diff
@@ -0,0 +1,248 @@
+    result: resources,
+    success: true
+  });
+});
+
+// get list of resources filtered by location radius
+router.get(
+  "/filter",
+  celebrate({
+    query: {
+      radius: Joi.number(),
+      lat: Joi.number(),
+      long: Joi.number(),
+      keyword: Joi.string()
+    }
+  }),
+  async (req, res) => {
+    const radius = req.query.radius;
+    const lat = req.query.lat;
+    const long = req.query.long;
+    const keyword = req.query.keyword;
+
+    let resources = await Resource.find({});
+
+    // 3963.2 = radius of Earth in miles
+    if (radius && lat && long) {
+      resources = await Resource.find({
+        location: {
+          $geoWithin: { $centerSphere: [[long, lat], radius / 3963.2] }
+        }
+      });
+
+      // sort by closest distance first
+      resources.sort(function(a, b) {
+        const distanceA = geolib.getDistance(
+          {
+            latitude: a.location.coordinates[0],
+            longitude: a.location.coordinates[1]
+          },
+          { latitude: lat, longitude: long }
+        );
+        const distanceB = geolib.getDistance(
+          {
+            latitude: b.location.coordinates[0],
+            longitude: b.location.coordinates[1]
+          },
+          { latitude: lat, longitude: long }
+        );
+
+        if (distanceA < distanceB) {
+          return -1;
+        }
+        return 1;
+      });
+    }
+    if (keyword) {
+      // fuzzy search
+      var options = {
+        shouldSort: true,
+        threshold: 0.5,
+        location: 0,
+        distance: 100,
+        maxPatternLength: 32,
+        minMatchCharLength: 1,
+        keys: [
+          {
+            name: "tags",
+            weight: 0.5
+          },
+          {
+            name: "companyName",
+            weight: 0.3
+          },
+          {
+            name: "description",
+            weight: 0.1
+          },
+          {
+            name: "address",
+            weight: 0.1
+          }
+        ]
+      };
+      const fuse = new Fuse(resources, options);
+      resources = fuse.search(keyword);
+    }
+
+    res.json({
+      code: 200,
+      result: resources,
+      success: true
+    });
+  }
+);
+
+// create new resource
+router.post(
+  "/",
+  celebrate({
+    body: Joi.object().keys({
+      companyName: Joi.string().required(),
+      contactName: Joi.string().required(),
+      contactPhone: Joi.string().required(),
+      contactEmail: Joi.string().required(),
+      description: Joi.string().required(),
+      address: Joi.string().required(),
+      location: Joi.object({
+        type: Joi.string().default("Point"),
+        coordinates: Joi.array()
+          .length(2)
+          .items(Joi.number())
+      }).required(),
+      notes: Joi.string(),
+      tags: Joi.array()
+        .items(Joi.string())
+        .required()
+    })
+  }),
+  async (req, res) => {
+    const data = req.body;
+    const newResource = new Resource({
+      companyName: data.companyName,
```

Comment:
We can actually reduce this to just `new Resource(data)` since all the fields are the same name as far as I can tell within the `Resource` model and what's being passed in.